core/src/main/clojure/xtdb/node/impl.clj

```diff
@@ -111,8 +113,13 @@
 
   (open-sql-query [this query query-opts]
     (let [query-opts (-> query-opts (with-query-opts-defaults this))]
-      (-> (xtp/prepare-sql this query query-opts)
-          (then-execute-prepared-query allocator query-timer query-opts))))
+      (try
+        (-> (xtp/prepare-sql this query query-opts)
+            (then-execute-prepared-query allocator query-timer query-opts))
+        (catch Exception e
+          (when query-error-counter
+            (.increment ^Counter query-error-counter))
+          (throw e)))))
 
   (open-xtql-query [this query query-opts]
     (let [query-opts (-> query-opts (with-query-opts-defaults this))]
```

Comment:
Even though this is currently not priority. Could we add the metrics also for XTQL.